### ISSUE TITLE
Incorrect deprecated_member_use_from_same_package in SDK edge

### ISSUE THREAD

**Issue by kevmoo** [dart-lang/linter#1393]
SDK @ https://github.com/dart-lang/sdk/commit/ec5ba8a2521a3638c39cc41a09ce736ed64cd092
Linter DEP @ 0.1.79

Build repo @ https://github.com/dart-lang/build/commit/9a6652ecb1ded9b916cc504512610db7fa2d5772

```console
$ cd build_modules
$ dartanalyzer .
Analyzing build_modules...
  hint • 'analyzer' is deprecated and shouldn't be used at lib/src/errors.dart:8:1 • deprecated_member_use_from_same_package
  hint • 'analyzer' is deprecated and shouldn't be used at lib/src/module_library.dart:8:1 • deprecated_member_use_from_same_package
2 hints found.
```

These lines are ignored in source with `// ignore: deprecated_member_use`

`pkg:analyzer` != `pkg:build_modules`

**Comment by kevmoo** [comment]
> The ignore comments need to be updated to use the new lint name.

But they are ignoring another package. The lint is a false positive!

**Comment by srawlins** [comment]
I can't reproduce this on a Dart synced and compiled just a few minutes ago ☹️ 

```
$ ~/code/dart-sdk/sdk/xcodebuild/ReleaseX64/dart-sdk/bin/dartanalyzer --version
dartanalyzer version 2.2.0-edge.b34ad87714c9203f998674c86635c2181be9a3da
$ ~/code/dart-sdk/sdk/xcodebuild/ReleaseX64/dart-sdk/bin/dartanalyzer lib/src/errors.dart           
Analyzing lib/src/errors.dart...
No issues found!
# After removing that `// ignore:`:
$ ~/code/dart-sdk/sdk/xcodebuild/ReleaseX64/dart-sdk/bin/dartanalyzer lib/src/errors.dart 
Analyzing lib/src/errors.dart...
  hint • 'analyzer' is deprecated and shouldn't be used at lib/src/errors.dart:7:1 • deprecated_member_use
1 hint found.
```

**Comment by kevmoo** [comment]
Try running analyzer just in the root of the build_modules package?

**Comment by srawlins** [comment]
Same.

```
$ ~/code/dart-sdk/sdk/xcodebuild/ReleaseX64/dart-sdk/bin/dartanalyzer .                             
Analyzing build_modules...
  hint • 'analyzer' is deprecated and shouldn't be used at lib/src/errors.dart:7:1 • deprecated_member_use                                    
1 hint found.
```

I have not run any build stuff in my checkout, so my `.dart_tool/` is woefully missing a build/ directory. Is yours?

```
$ ls .dart_tool/
pub
```

**Comment by kevmoo** [comment]
Surprise, surprise: if I delete `.dart_tool/build` it all works as expected.

**Comment by srawlins** [comment]
erp, I can reproduce now with that directory. Looking into this...

**Comment by srawlins** [comment]
Fix in the mail: https://dart-review.googlesource.com/c/sdk/+/91441

**Comment by kevmoo** [comment]
Thanks!

**Comment by kevmoo** [comment]
This was supposed to be fixed, but I'm seeing again in 2.3.1-dev.0.0

Sync and load https://github.com/dart-lang/pubspec_parse/

```
Analyzing pubspec_parse...
  hint • 'author' is deprecated and shouldn't be used at test/parse_test.dart:19:18 • deprecated_member_use
  hint • 'author' is deprecated and shouldn't be used at test/parse_test.dart:52:18 • deprecated_member_use
  hint • 'author' is deprecated and shouldn't be used at test/parse_test.dart:128:20 • deprecated_member_use
  hint • 'author' is deprecated and shouldn't be used at test/parse_test.dart:138:20 • deprecated_member_use
  hint • 'author' is deprecated and shouldn't be used at test/parse_test.dart:148:20 • deprecated_member_use
  hint • 'author' is deprecated and shouldn't be used at test/parse_test.dart:159:20 • deprecated_member_use
  hint • 'author' is deprecated and shouldn't be used at test/parse_test.dart:170:20 • deprecated_member_use
7 hints found.
```

CC @srawlins

**Comment by srawlins** [comment]
Apologies, @kevmoo . I've found the issue and have a fix in the works; I can mail tomorrow.

**Comment by kevmoo** [comment]
No worries! Thanks.

*From: *Sam Rawlins <notifications@github.com>
*Date: *Thu, May 9, 2019 at 9:17 PM
*To: *dart-lang/linter
*Cc: *Kevin Moore, Mention

Apologies, @kevmoo <https://github.com/kevmoo> . I've found the issue and
